### DISCUSSION TITLE
Spinning up code-server inside a WebContainer

### DISCUSSION THREAD

**Discussion by wpitallo** [coder/code-server#3485] [Ideas]
Hey, not sure if you have seen what the guys at StackBlitz have been busy with. They have this concept of WebContainer that essentially runs only inside a browser. Imagine it would be possible to launch code-server  from lets say a server less function that then spins up an instance of code-server inside the browsers without requiring and infrastructure.  Then it may even be possible to created customized code-server (dev templates) that could be initialized and loaded in the users browser via http links. 
🤯
🤯
🤯

**Comment by jsjoeio** [comment]
Sounds a bit like VSCode + Cloudflare Workers. Have you seen [this project](https://github.com/glenstack/vscode-workers)?

This is a really fascinating idea though. cc @bpmct

**Reply by wpitallo** [reply]
That does look promising, does not seem like you can actually build or access a container environment from that example tho but yea if you could now access a container env with node runtime running inside the javascrpit VM that would be amazing. Then have that connected to GitHub repo etc could be awesome. Will be great to be able to have the VS code plugins / extensions etc preconfigured as well upon boot of the project / container. Maybe this is what Github codespaces is going to do: https://github.com/features/codespaces. Very exciting :)

**Reply by jsjoeio** [reply]
I wish the StackBlitz Web Containers were open-source! I'm really curious how they implemented everything. That all would be awesome! Maybe it'll be easy to implement in the near future :)

**Reply by wpitallo** [reply]
Yea its a massive bummer that they not, would also love to know how it all fits together.
